### ISSUE TITLE
Do not reinstall already installed rock

### ISSUE THREAD

**Issue by deathbeam** [keplerproject/luarocks#417]
When using for exampe

```lua
luarocks install ldoc
```

it will install latest version. But when running it again, it simply reinstalls it. This can be fixed with few lines of code to check if rock is already installed.

**Comment by Tieske** [comment]
I know this behaviour, but to me this is a feature. It makes it easy to fix stuff after screwing up.

Also; it only does this when you explicitly use the `install` command, but not when installing dependencies. In the latter case it will check and only install if it is missing.

**Comment by deathbeam** [comment]
Yes I know about that dependency thing. But why would you need to fix stuff in this way? You can just do remove and then install. In all other package managers, when something is installed, it will not reinstall it every time user tries to install it. This `install` thing is big problem when you want to integrate LuaRocks in other build automation tool (like I am doing in my personal project). Also, when you have rock already installed and you use the `install` command, you do not have chance to even notice that that rock was previously installed on system, because it will reinstall it immediately without any warning or notice.

In short, in my opinion this is serious bug and not feature at all (or very implicit , specific and hidden feature).

**Comment by hishamhm** [comment]
I'm on the fence about this. 

@Tieske, how would you feel if the behavior was like this?

```
$ luarocks install foo
Checking latest version of foo...
foo 1.0-1 is the latest version and it is already installed. To force a reinstall, use --force.
$ luarocks install foo --force
Checking latest version of foo...
Installing foo 1.0-1...
...
```

**Comment by ignacio** [comment]
@hishamhm There is also what was discussed here:
http://sourceforge.net/p/luarocks/mailman/message/33642794/

**Comment by Tieske** [comment]
using  `--force` would be ok by me (it's not that big a deal anyway, just something we're used to). As for the message @Ignacio pointed out, I'd go for option A.

**Comment by siffiejoe** [comment]
The `--force` suggestion in this thread is fine by me. For the other options mentioned on the mailing list post I'll reply there (spoiler: I'd go for C) ...

**Comment by mpeterv** [comment]
Original issue with some more discussion: #22
